### ISSUE TITLE
Updating any of the compositeId fields does not persist in database

### ISSUE THREAD

**Issue by snimavat** [grails/grails-data-mapping#1144]
When there is a domain with composite Id, and if one of the field in compositeId is changed, saving the domain (even with flush) will not persist the change to database.

Grails: 3.3.3
Also reproducible with Gorm 6.1.11.BUILD-SNAPSHOT

Attached sample has a **CompositeIdSpec** which reveals the issue
[bugwork-composite-id.zip](https://github.com/grails/grails-data-mapping/files/2297376/bugwork-composite-id.zip)

**Comment by jameskleeh** [comment]
@snimavat Normally it's not a good idea to change the primary key. If you want a record with a different primary key you should delete the first one and re-insert.

**Comment by snimavat** [comment]
@jameskleeh I agree, however this used to work, and not now, so probably some thing related to gorm and not hibernate... even if this is not a good practice, can we consider, that this is a bug, should it work or not ?

**Comment by snimavat** [comment]
Any update on this one !? is it planned to fix !
